bson/bsoncodec/registry.go

```diff
@@ -322,14 +322,30 @@ func (r *Registry) LookupEncoder(t reflect.Type) (ValueEncoder, error) {
 		if enc == nil {
 			return nil, ErrNoEncoder{Type: t}
 		}
-		return enc, nil
+		// if dec is a condAddrEncoder, we still need to fill in elseEnc
```

Comment:
I'm not sure I understand this. Let's say I have type `T` and `*T` implements `Marshaler`. The first time I try to marshal value of type `T`, this should call `lookupInterfaceEncoder`, which will return:
```
condAddrEncoder{
    canAddrEnc: MarshalerEncodeValue
    elseEnc: StructCodec.EncodeValue
}
```
and we'll store that in `typeEncoders` as a cached entry. The next time I try to marshal something of type `T`, why do we need to go through that whole process again? Are there cases where the cached `condAddrEncoder` is not correct?